GitUI/UserControls/BlameControl.cs

```diff
@@ -641,8 +646,59 @@ private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventAr
             // Try get actual parent revision, get popup if it does not exist.
             // (The menu should be disabled if previous is not in grid).
             GitRevision? revision = _revisionGridInfo?.GetActualRevision(blameInfo.selectedRevision);
-            _clickedBlameLine = _lastBlameLine;
-            BlameRevision(revision?.FirstParentId, blameInfo.filename);
+            ObjectId? parentId = revision?.FirstParentId;
+
+            int selectedLine = _lastBlameLine!.OriginLineNumber;
+            int calculatedLine = selectedLine;
+
+            if (selectedLine != 1 && parentId != null)
+            {
+                try
+                {
+                    string currentLineText = _lastBlameLine.Text.Split("\n").First();
+
+                    // Get the git diff of changes introduced by the blamed selected commit
+                    // *without context lines* to make starting line value more accurate and possibly have more distincts chunks!
+                    GitArgumentBuilder args = new("diff") { $"-U0", parentId, blameInfo.selectedRevision.ObjectId, "--", blameInfo.filename };
+
+                    ExecutionResult executionResult = UICommands.Module.GitExecutable.Execute(args);
+
+                    string diffOutput = executionResult.StandardOutput;
```

Comment:
Let it handle the exit code regardless of with or without `cache`:
```suggestion
                    string diffOutput = UICommands.Module.GitExecutable.GetOutput(args, cache: GitModule.GitCommandCache);
```
Since the output of the command will not change, we can cache it and save a git call for the case the user goes back and forth.

Reply:
I don't know how this command cache is managed and if the cache size is limited by a quota but even if yes the same blame command could be run multiple times by the user, I don't think it will be done very often. I prefer not to fill the memory with that (except if you tell me that the cache is cleaned some time to time).

Reply:
OK, I had a quick look at the cache. Not 100% convinced but may be OK. So, done.